test/boost/secondary_index_test.cc

```diff
@@ -1208,6 +1208,279 @@ SEASTAR_TEST_CASE(test_indexing_paging_and_aggregation) {
     });
 }
 
+// Verifies that both "SELECT * [rest_of_query]" and "SELECT count(*) [rest_of_query]" 
+// return expected count of rows.
+void assert_select_count_and_select_rows_has_size(cql_test_env& e, const sstring& rest_of_query, int64_t expected_count) {
+    eventually([&] { 
+        auto res = e.execute_cql("SELECT count(*) " + rest_of_query).get0();
+        assert_that(res).is_rows().with_rows({
+            { long_type->decompose(expected_count) }
+        });
+        res = e.execute_cql("SELECT * " + rest_of_query).get0();
+        assert_that(res).is_rows().with_size(expected_count);
+    });
+}
+
+static constexpr int page_scenarios_row_count = 2 * cql3::statements::select_statement::DEFAULT_COUNT_PAGE_SIZE + 120;
+static constexpr int page_scenarios_initial_count = 3;
+static constexpr int page_scenarios_window_size = 4;
+static constexpr int page_scenarios_just_before_first_page = cql3::statements::select_statement::DEFAULT_COUNT_PAGE_SIZE - page_scenarios_window_size;
+static constexpr int page_scenarios_just_after_first_page = cql3::statements::select_statement::DEFAULT_COUNT_PAGE_SIZE + page_scenarios_window_size;    
+static constexpr int page_scenarios_just_before_second_page = 2 * cql3::statements::select_statement::DEFAULT_COUNT_PAGE_SIZE - page_scenarios_window_size;
+static constexpr int page_scenarios_just_after_second_page = 2 * cql3::statements::select_statement::DEFAULT_COUNT_PAGE_SIZE + page_scenarios_window_size;    
+
+static_assert(page_scenarios_initial_count < page_scenarios_row_count);
+static_assert(page_scenarios_window_size < cql3::statements::select_statement::DEFAULT_COUNT_PAGE_SIZE);
+static_assert(page_scenarios_just_after_second_page < page_scenarios_row_count);
+
+// Executes `insert` lambda page_scenarios_row_count times. 
+// Runs `validate` lambda in a few scenarios:
+//
+// 1. After a small number of `insert`s
+// 2. In a window from just before and just after `insert`s where executed
+//    DEFAULT_COUNT_PAGE_SIZE times
+// 3. In a window from just before and just after `insert`s where executed
+//    2 * DEFAULT_COUNT_PAGE_SIZE times
+// 4. After all `insert`s
+void test_with_different_page_scenarios(
+    noncopyable_function<void (int64_t)> insert, noncopyable_function<void (int64_t)> validate) {
+
+    int64_t current_row = 0;
```

Comment:
Why `int64_t`?  Can it be negative?  Can it exceed 32 bits?  Perhaps an explaining comment is needed.

Reply:
There was no particular reason to use `int64_t` (probably as I was writing this I was influenced by the fact that `COUNT(*)` returns `long` - `int64`). 

Changed the type to `int` to indicate to the readers of the code that this is just a "normal number" - a counter that will not exceed a few tens of thousands in this case.